web/src/UserEditPage.js

```diff
@@ -104,8 +104,9 @@ class UserEditPage extends React.Component {
 
   setReturnUrl() {
     const searchParams = new URLSearchParams(this.props.location.search);
-    const returnUrl = searchParams.get("returnUrl");
-    returnUrl !== null ? this.setState({returnUrl: returnUrl}) : null;
+    this.setState({
+      returnUrl: searchParams.get("returnUrl"),
```

Comment:
The thirty-party app call API get-account. It will first through the frontend route of casdoor. 
- This time the component can get the `returnUrl` and setState.  
- Then the casdoor frontend call server api. It cause the `access_token` and `returnUrl` disappear.  setState will set `returnUrl = null`. 

So if the `returnUrl` is null. It should  not modify the state

Reply:
Is it OK now?

Reply:
Yes, it is ok now.